src/probnum/problems/_problems.py

```diff
@@ -1 +1,204 @@
+        Likelihood of the observations; that is, relation between the latent process and the observed values.
+        Encodes for example noise.
+    solution
+        Closed form, analytic solution to the problem. Used for testing and benchmarking.
+
+    Examples
+    --------
+    >>> obs = [11.4123, -15.5123]
+    >>> loc = [0.1, 0.2]
+    >>> rp = RegressionProblem(observations=obs, locations=loc)
+    >>> rp
+    RegressionProblem(observations=[11.4123, -15.5123], locations=[0.1, 0.2], likelihood=None, solution=None)
+    >>> rp.observations
+    [11.4123, -15.5123]
+    """
+
+    observations: np.ndarray
+    locations: np.ndarray
+
+    # Optional, because it should be specifiable without explicit likelihood info.
+    likelihood: typing.Optional[pnfs.statespace.DiscreteGaussian] = None
+
+    # For testing and benchmarking
+    solution: typing.Optional[
+        typing.Callable[[np.ndarray], typing.Union[float, np.ndarray]]
+    ] = None
+
+
+@dataclasses.dataclass
+class IVProblem:
+    r"""First order ODE initial value problem.
+
+    Compute a function :math:`y=y(t)` that solves
+
+    .. math::
+        \dot y(t) = f(t, y(t)), \quad y(t_0) = y_0
+
+    on time-interval :math:`[t_0, t_\text{max}]`.
+    Solved by the ODE solvers in :mod:`probnum.diffeq`.
+
+
+    Parameters
+    ----------
+    f
+        ODE vector-field.
+    t0
+        Initial point in time.
+    tmax
+        Final point in time.
+    y0
+        Initial value of the solution.
+    df
+        Jacobian of the ODE vector-field :math:`f=f(t,y)` with respect to the :math:`y` variable.
+    ddf
+        Hessian of the ODE vector-field :math:`f=f(t,y)` with respect to the :math:`y` variable.
+    solution
+        Closed form, analytic solution to the problem. Used for testing and benchmarking.
+
+    Examples
+    --------
+    >>> def f(t, x):
+    ...     return x*(1-x)
+    >>> ivp = IVProblem(f, t0=0., tmax=3., y0=0.1)
+    >>> ivp.t0, ivp.tmax, ivp.y0
+    (0.0, 3.0, 0.1)
+    >>> np.round(ivp.f(ivp.t0, ivp.y0), 2)
+    0.09
+    """
+
+    f: typing.Callable[[float, np.ndarray], np.ndarray]
+    t0: float
+    tmax: float  # Bold move: remove this? This is not really part of an IVP.
+    y0: typing.Union[pntp.FloatArgType, np.ndarray]
+    df: typing.Optional[typing.Callable[[float, np.ndarray], np.ndarray]] = None
+    ddf: typing.Optional[typing.Callable[[float, np.ndarray], np.ndarray]] = None
+
+    # For testing and benchmarking
+    solution: typing.Optional[typing.Callable[[float, np.ndarray], np.ndarray]] = None
+
+
+@dataclasses.dataclass
+class LinearSystem:
+    r"""Linear system of equations.
+
+    Compute :math:`x` from :math:`Ax=b`.
+    Solved by the probabilistic linear solver in :mod:`probnum.linalg`
+
+    Parameters
+    ----------
+    A
+        Square matrix or linear operator.
+    b
+        Right-hand side vector or matrix.
+    solution
+        Closed form, analytic solution to the problem. Used for testing and benchmarking.
```

Comment:
For linear systems "closed form" doesn't really make sense. I think "True solution" or "Analytic solution" is a better naming for `LinearSystem`.

Reply:
Let's try to put this information everywhere then, by perhaps renaming the `solution` field into `analytical_solution`? I find it okay if it is cumbersome to call, we will only need it rarely (and mostly internally).

Reply:
I think `solution` is good, it's just the docstring that doesn't match here. Easy fix would be to just drop "Closed form" from the docstring.